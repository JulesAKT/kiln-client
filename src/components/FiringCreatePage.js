import React from "react";
import { useDispatch } from "react-redux";
import { createFiring } from "../actions";
import FiringForm from "./FiringForm";

const FiringCreatePage = (props) => {
  const dispatch = useDispatch();

  const handleSubmit = (formValues) => {
    const values = {
      project_id: props.match.params.id,
      order: props.match.params.order,
      ...formValues,
    };
    console.log("submitting");
    console.log(values);
    dispatch(createFiring(values));
  };

  return (
    <div>
      <FiringForm
        onSubmit={handleSubmit}
        initialValues={{
          notes: "",
          name: "",
          date: Date.now(),
        }}
      />
    </div>
  );
};

export default FiringCreatePage;
