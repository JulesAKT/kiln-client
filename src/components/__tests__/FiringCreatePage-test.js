import React from "react";
import Root from "../../root";
import FiringCreatePage from "../FiringCreatePage";

import { waitForElement, render } from "@testing-library/react";

it("FiringCreatePage should render against a snapshot", async () => {
  const { container } = render(
    <Root>
      <FiringCreatePage match={{ params: { id: "2" } }} />
    </Root>
  );
  expect(container).toMatchSnapshot();
});
