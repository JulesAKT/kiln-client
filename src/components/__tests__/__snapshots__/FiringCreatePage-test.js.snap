// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FiringCreatePage should render against a snapshot 1`] = `
<div>
  <div>
    <div
      class="ui segment"
    >
      <form
        class="ui form"
      >
        <div
          class="field undefined "
        >
          <div
            class="field"
          >
            <label>
              Firing Name
            </label>
            <div
              class="ui input"
            >
              <input
                name="name"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="field"
        >
          <label>
            Firing Date
          </label>
          <div
            class="react-datepicker-wrapper"
          >
            <div
              class="react-datepicker__input-container"
            >
              <input
                class=""
                type="text"
                value="31 Dec 2021"
              />
            </div>
          </div>
        </div>
        <div
          class="field undefined "
        >
          <div
            class="field"
          >
            <label>
              Notes
            </label>
            <textarea
              name="notes"
              rows="8"
            />
          </div>
        </div>
        <button
          class="ui disabled button"
          disabled=""
          tabindex="-1"
        >
          Update
        </button>
      </form>
    </div>
  </div>
</div>
`;
